### PR TITLE
Intl.NumberFormat. v3 fixes

### PR DESCRIPTION
**Pull request by hamishwillee**
FF116 adds support for Intl.NumberFormat v3 proposal. This has largely been documented already, but this fixes some omissions/does some tidy up. The work might happen across a number of PRs.

Fixes in this one:
- `Intl.NumberFormat` constructor 
  - remove the experimental inline markers for the new options.
  - Add some exceptions (might not be exhaustive - but is something.)
- `NumberFormat.resolvedOptions()` 
  - new options for trailing zeros and rounding might be returned (in addition to the `userGrouping` already indicated. These have been added.
  - Added missing results: `unit`, `unitDisplay`
  - Added cross links to the constructor docs, which define the values you might receive properly - i.e. this is just a summary - if you want detail you look in the constructor.
  - Improved the example to show more of these new options.

Other docs work can be tracked in https://github.com/mdn/content/issues/27746

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/intl/pluralrules/selectrange/index.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ selectRange(startRange, endRange)
 ### Return value
 
 A string representing the pluralization category of the `number`.
-This can be one of `zero`, `one`, `two`, `few`, `many` or `other`, that are relevant for the locale whose localization is specified in [LDML Language Plural Rules](https://unicode-org.github.io/cldr-staging/charts/37/supplemental/language_plural_rules.html#rules).
+This can be one of `zero`, `one`, `two`, `few`, `many` or `other`, that are relevant for the locale whose localization is specified in [LDML Language Plural Rules](https://www.unicode.org/cldr/charts/43/supplemental/language_plural_rules.html).
```
Comment (hamishwillee) [inline files/en-us/web/javascript/reference/global_objects/intl/pluralrules/selectrange/index.md:21]: FYI Link moved.

File: files/en-us/web/javascript/reference/global_objects/intl/numberformat/resolvedoptions/index.md [hunk 2]
```diff
@@ -55,18 +81,36 @@ Only one of the following two groups of properties is included:
...
-usedOptions.notation; // "standard"
-usedOptions.signDisplay; // "auto"
-usedOptions.style; // "decimal"
-usedOptions.minimumIntegerDigits; // 1
-usedOptions.minimumFractionDigits; // 0
-usedOptions.maximumFractionDigits; // 3
-usedOptions.useGrouping; // true
+console.log(usedOptions.locale); // "de-DE"
+console.log(usedOptions.numberingSystem); // "latn"
+console.log(usedOptions.compactDisplay); // "undefined"
```
Comment (hamishwillee) [inline files/en-us/web/javascript/reference/global_objects/intl/numberformat/resolvedoptions/index.md:97]:
```suggestion
console.log(usedOptions.compactDisplay); // undefined ("notation" not set to "compact")
```

**Comment by hamishwillee** [comment]
Note, there is more to be done here, but I'm away until Friday and most of that work will be in PluralRules - so we might as well get this in if possible.

**Review by Josh-Cena — APPROVED** [review]
Constructor change LGTM; I would trust you with the `resolvedOptions` change because I plan to homogenize/deduplicate all the `resolvedOptions` pages later anyway ^^

**Comment by hamishwillee** [comment]
Thanks very much.

**Pull request merged**